clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -808,7 +808,8 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                 }
                 log.debug("Subscribed to topic(s): {}", Utils.join(topics, ", "));
                 this.subscriptions.subscribe(topics, listener);
-                metadata.setTopics(subscriptions.groupSubscription());
+                this.metadata.needMetadataForAllTopics(true);
```

Comment:
@hachikuji I don't think that actually would work now, because we are not immediately updating subscribed topics anymore (the reason the KIP started). So even if we trigger a rebalance assigned partitions won't change until metadata gets updated. That tells me we need to update metadata in any case. Do you agree?